app/src/main/java/woowacourse/shopping/view/cart/selection/CartProductSelectionViewModelFactory.kt

```
@@ -0,0 +1,17 @@
+import androidx.lifecycle.ViewModelProvider
+import woowacourse.shopping.domain.repository.CartProductRepository
+
+class CartProductSelectionViewModelFactory(
+    private val cartProductRepository: CartProductRepository,
+) : ViewModelProvider.Factory {
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(CartProductSelectionViewModel::class.java)) {
+            return CartProductSelectionViewModel(cartProductRepository) as T
+        }
+        throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)